src/main/java/org/craftercms/core/service/impl/ContentStoreServiceImpl.java

```diff
@@ -143,29 +143,42 @@ public Context getContext(String contextId) {
      * {@inheritDoc}
      */
     @Override
-    public Context createContext(String storeType, String storeServerUrl, String username, String password, String rootFolderPath,
-                                 boolean mergingOn, boolean cacheOn, int maxAllowedItemsInCache,
-                                 boolean ignoreHiddenFiles) throws InvalidStoreTypeException, RootFolderNotFoundException, StoreException,
-        AuthenticationException {
-        String id = createContextId(storeType, storeServerUrl, username, password, rootFolderPath, cacheOn,
-                                    maxAllowedItemsInCache, ignoreHiddenFiles);
+    @Deprecated
+    public Context createContext(String storeType, String storeServerUrl, String username, String password,
+                                 String rootFolderPath, boolean mergingOn, boolean cacheOn, int maxAllowedItemsInCache,
+                                 boolean ignoreHiddenFiles)
+        throws InvalidStoreTypeException, RootFolderNotFoundException, StoreException, AuthenticationException {
+        return getContext(null, storeType, rootFolderPath, mergingOn, cacheOn, maxAllowedItemsInCache,
+            ignoreHiddenFiles);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Context getContext(final String ownerId, final String storeType, final String rootFolderPath,
+                              final boolean mergingOn, final boolean cacheOn, final int maxAllowedItemsInCache,
+                              final boolean ignoreHiddenFiles)
+        throws InvalidStoreTypeException, RootFolderNotFoundException, StoreException, AuthenticationException {
+        String id =
+            createContextId(ownerId, storeType, rootFolderPath, cacheOn, maxAllowedItemsInCache, ignoreHiddenFiles);
 
         if (!contexts.containsKey(id)) {
             ContentStoreAdapter storeAdapter = storeAdapterRegistry.get(storeType);
             if (storeAdapter == null) {
                 throw new InvalidStoreTypeException("No registered content store adapter for store type " + storeType);
             }
 
-            Context context = storeAdapter.createContext(id, storeServerUrl, username, password, rootFolderPath,
-                                                         mergingOn, cacheOn, maxAllowedItemsInCache, ignoreHiddenFiles);
+            Context context = storeAdapter.createContext(id, null, null, null, rootFolderPath,
+                mergingOn, cacheOn, maxAllowedItemsInCache, ignoreHiddenFiles);
 
             cacheTemplate.getCacheService().addScope(context);
 
             contexts.put(id, context);
 
             return context;
         } else {
-            throw new StoreException("A context for id '" + id + "' already exists");
+            return contexts.get(id);
```

Comment:
Please keep the old implementation and let it throw the already exists exception. The old implementation shouldn't change at all. It can use the new createId with `null` as the first param, that's fine.